test/nuts/scale/lotsOfClasses.nut.ts

```diff
@@ -0,0 +1,115 @@
+    for (let d = 0; d < dirCount; d++) {
+      const dirName = path.join(classdir, `dir${d}`);
+      // eslint-disable-next-line no-await-in-loop
+      await fs.promises.mkdir(dirName);
+      for (let c = 0; c < classesPerDir; c++) {
+        const className = `x${d}x${c}`;
+        // eslint-disable-next-line no-await-in-loop
+        await Promise.all([
+          fs.promises.writeFile(
+            path.join(dirName, `${className}.cls`),
+            `public with sharing class ${className} {public ${className}() {}}`
+          ),
+          fs.promises.writeFile(
+            path.join(dirName, `${className}.cls-meta.xml`),
+            '<?xml version="1.0" encoding="UTF-8"?><ApexClass xmlns="http://soap.sforce.com/2006/04/metadata"><apiVersion>54.0</apiVersion><status>Active</status></ApexClass>'
+          ),
+        ]);
+      }
+    }
+  });
+
+  after(async () => {
+    await recordPerf(testName, performance);
+    await session?.clean();
+  });
+
+  it('componentSetBuilder', async () => {
+    performance.mark('ComponentSetBuild');
+
+    await ComponentSetBuilder.build({
+      sourcepath: [session.project.dir],
+    });
+    performance.measure('componentSetCreate', 'ComponentSetBuild');
+  });
+
+  it('convert source to mdapi', async () => {
+    const resolver = new MetadataResolver();
+    const converter = new MetadataConverter();
+
+    performance.mark('SourceToMdapi');
+
+    const components = resolver.getComponentsFromPath(path.join(session.project.dir, 'force-app'));
+    await converter.convert(components, 'metadata', {
+      type: 'directory',
+      outputDirectory: path.join(session.project.dir, 'mdapiOut'),
+      packageName: 'MetadataFormatPackage',
+    });
+    performance.measure('sourceToMdapi', 'SourceToMdapi');
+  });
+  it('convert source to zip', async () => {
+    const resolver = new MetadataResolver();
+    const converter = new MetadataConverter();
+
+    performance.mark('SourceToZip');
+
+    const components = resolver.getComponentsFromPath(path.join(session.project.dir, 'force-app'));
+    await converter.convert(components, 'metadata', {
+      type: 'zip',
+      outputDirectory: path.join(session.project.dir, 'mdapiOut'),
+      packageName: 'MetadataFormatPackage',
+    });
+    performance.measure('sourceToZip', 'SourceToZip');
+  });
+  it('convert mdapi to source', async () => {
+    const resolver = new MetadataResolver();
+    const converter = new MetadataConverter();
+
+    performance.mark('MdapiToSource');
+
+    const components = resolver.getComponentsFromPath(path.join(session.project.dir, 'mdapiOut'));
+    await converter.convert(components, 'source', {
+      type: 'directory',
+      outputDirectory: path.join(session.project.dir, 'mdapiOut'),
+      packageName: 'SourceFormatPackage',
+    });
+    performance.measure('mdapiToSource', 'MdapiToSource');
+  });
+  it('convert zip to source', async () => {});
```

Comment:
Did you mean to implement this one?

Reply:
I left it for feedback.  I'm not sure what's interesting to do now vs. later vs. not (vs. getting some baselines and moving on to the actual problem this is all pre-work for)

Reply:
digging around sdr, it looks like unzip streams to convert anyway (like an alternate reader).  I'll remove for now.